File: 2432-number-of-zero-filled-subarrays/2432-number-of-zero-filled-subarrays.cpp
class Solution {
public:
    long long zeroFilledSubarray(vector<int>& nums) {
        
        long long int ans=0;
        for(int i=0;i<nums.size();i++)
        {
            if(nums[i]!=0)
              continue;
            long long int res =0;
            long long int len =0;
            
                res=1;
                 len=1;
                 i++;
                while(i<nums.size() && nums[i]==0)
                {
                   len = len + 1;
                   res +=len;
                   i++;
                }
            
            ans += res;
        }
        return ans;
    }
};

File: 2432-number-of-zero-filled-subarrays/2432-number-of-zero-filled-subarrays.py
class Solution:
    def zeroFilledSubarray(self, nums: List[int]) -> int:
        # ans , i = 0 , 0
        # while i <len(nums):
        #     if not nums[i]==0:
        #         i+=1
        #         continue
        #     res=1
        #     leng =1
        #     i+=1
        #     while i < len(nums) and nums[i]==0:
        #         leng +=1
        #         res+= leng
        #         i += 1
        #     ans += res    
        #     i+=1

        #simpler code
        ans,leng  = 0,0
        for i in range(len(nums)):
            if nums[i] == 0:
                leng+=1
            else:
                leng = 0
            ans+=leng         
        return ans